adoc/chapters/architecture.adoc

```diff
@@ -1758,6 +1758,7 @@ code as well as the <<device-image, device images>> for the
 As in the <<smcp>> case, each <<device-image>> could either contain native
 device ISA or an intermediate language.
 
+[[subsec:library-only]]
```

Comment:
I can of course outline these changes into a separate PR. I needed more anchors to avoid harcoding section names and I still need a few more of them (for `vec` and `marray` aliases, in particular)